### PR TITLE
test: e2e test

### PR DESCRIPTION
**Pull request by JeyJeyGao**
Test:
- added e2e test for Linux, windows and macOS
- includes 14 test cases: imported pem/pfx certs chain, unordered pem/pfx certs chain, azure key vault csr pem/pfx certs chain, self-signed pem/pfx certs, partial pem/pfx certs chain with local cert bundle and corresponding invalid cases.
- the E2E automatically run for PRs from `Azure/notation-azure-kv` repo and will be skipped for folked repo

Additional files:
- dockerfile for build distribution registry on windows
- zot configuration for local registry on macOS
- generate-certs.sh script for create certificate on azure key vault for future re-generating certificates.

### CODE REVIEW

**Review by shizhMSFT — COMMENTED** [review]
Should `.github/e2e/` be put at `tests/e2e/`?

**Comment by JeyJeyGao** [comment]
> Should `.github/e2e/` be put at `tests/e2e/`
 
@shizhMSFT Updated.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -52,7 +49,147 @@ jobs:
...
+    steps:
+      - name: Check out code into the project directory
+        uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: linux-amd64-binary
+          path: ./bin/artifacts
+      - name: Run download server locally
+        run: |
+          nohup python3 ./test/e2e/download-server.py &
```
Comment (shizhMSFT) [inline .github/workflows/test.yml:94]: Is it possible to use `python3 -m http.server`?

Comment (JeyJeyGao) [reply .github/workflows/test.yml]: Awsome! Updated.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**